lib/web/integrations_azureoidc.go

```diff
@@ -0,0 +1,91 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package web
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/google/safetext/shsprintf"
+	"github.com/gravitational/trace"
+	"github.com/julienschmidt/httprouter"
+
+	"github.com/gravitational/teleport/lib/httplib"
+	"github.com/gravitational/teleport/lib/utils/oidc"
+	"github.com/gravitational/teleport/lib/web/scripts/oneoff"
+)
+
+// azureOIDCConfigureIdP returns a script that configures Azure OIDC Integration
+// by creating an Enterprise Application in the Azure account
+// with Teleport OIDC as a trusted credential issuer.
+func (h *Handler) azureOIDCConfigure(w http.ResponseWriter, r *http.Request, p httprouter.Params) (any, error) {
+	ctx := r.Context()
+	queryParams := r.URL.Query()
+
+	oidcIssuer, err := oidc.IssuerFromPublicAddress(h.cfg.PublicProxyAddr)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	authConnectorName := queryParams.Get("authConnectorName")
+	if authConnectorName == "" {
+		return nil, trace.BadParameter("authConnectorName must be specified")
+	}
+	// Ensure the auth connector name is valid
+	const withSecrets = false
+	_, err = h.GetProxyClient().GetSAMLConnector(ctx, authConnectorName, withSecrets)
+	// NotFound error is ignored to prevent disclosure of whether the integration exists in a public/no-auth endpoint.
+	if err != nil && !trace.IsNotFound(err) {
+		return nil, trace.Wrap(err)
+	}
+
+	integrationName := queryParams.Get("integrationName")
+	if len(integrationName) == 0 {
+		return nil, trace.BadParameter("integrationName must be specified")
+	}
+	// Ensure the integration name is valid.
+	_, err = h.GetProxyClient().GetIntegration(ctx, integrationName)
+	// NotFound error is ignored to prevent disclosure of whether the integration exists in a public/no-auth endpoint.
+	if err != nil && !trace.IsNotFound(err) {
```

Comment:
nit: You can skip `err != nil` as `nil` returns false from `trace.IsNotFound()`.

Reply:
`trace.IsNotFound(nil)` would return false, so `!trace.IsNotFound(nil)` would be true, and we'd return the error. But we don't want to early-return if the error is `nil`.